index.js

```diff
@@ -21,6 +22,25 @@ function paramify(list, prefix) {
 	return result;
 }
 
+function isThenable(obj) {
```

Comment:
Yes, it is useful, because I need to know whether a thing is a promise or not.

If you do not return a promise from a transaction, we lock down any subsequent calls to `query` on that transaction immediately, because we commit the transaction in the next `then` in the promise chain. Since promises are resolved asynchronously, if we didn't lock it down, it would be possible to call a `query` inside `transaction()` _after_ it has been committed, and we don't want that.

If you do return a promise from a transaction - the lock down happens after that promise is resolved, i.e. it is the consumer's responsibility to handle any possible race conditions.